reconcile/deadmanssnitch.py

```diff
@@ -0,0 +1,244 @@
+                self.deadmanssnitch_api.delete_snitch(token)
+            case UpdateVaultDiffData(
+                cluster_name=cluster_name, check_in_url=check_in_url
+            ):
+                self.vault_client.write(
+                    {
+                        "path": self.settings.snitches_path,
+                        "data": {f"deadmanssnitch-{cluster_name}-url": check_in_url},
+                    },
+                    decode_base64=False,
+                )
+
+    def create_snitch(self, cluster_name: str) -> None:
+        tags = ["app-sre"]
+        alert_email = [self.settings.alert_email]
+        payload = {
+            "name": f"prometheus.{cluster_name}.devshift.net",
+            "alert_type": "Heartbeat",
+            "interval": "15_minute",
+            "tags": tags,
+            "alert_email": alert_email,
+            "notes": self.settings.notes_link,
+        }
+
+        snitch = self.deadmanssnitch_api.create_snitch(payload=payload)
+        self.vault_client.write(
+            {
+                "path": self.settings.snitches_path,
+                "data": {f"deadmanssnitch-{cluster_name}-url": snitch.check_in_url},
+            },
+            decode_base64=False,
+        )
+
+
+class DeadMansSnitchIntegration(QontractReconcileIntegration[NoParams]):
+    """Integration to automate deadmanssnitch snitch api during cluster dressup."""
+
+    @property
+    def name(self) -> str:
+        return QONTRACT_INTEGRATION
+
+    def __init__(self) -> None:
+        super().__init__(NoParams())
+        self.qontract_integration_version = make_semver(0, 1, 0)
+
+    def add_vault_data(self, snitch: Snitch, snitch_secret_path: str) -> Snitch:
+        try:
+            full_secret_path = {
+                "path": snitch_secret_path,
+                "field": f"deadmanssnitch-{snitch.get_cluster_name()}-url",
+            }
+            snitch.vault_data = self.secret_reader.read(full_secret_path).strip()
+        except SecretNotFound:
+            snitch.vault_data = SECRET_NOT_FOUND
+        return snitch
+
+    def get_current_state(
+        self,
+        deadmanssnitch_api: DeadMansSnitchApi,
+        clusters: list[ClusterV1],
+        snitch_secret_path: str,
+    ) -> dict[str, Snitch]:
+        # current state includes for deadmanssnithch response and associated secret in vault
+        snitches = deadmanssnitch_api.get_snitches(tags=["app-sre"])
+        # create snitch_map only for  the desired clusters
+        snitches_with_cluster_mapping = {
+            snitch.get_cluster_name(): snitch for snitch in snitches
+        }
+        current_state = {
+            cluster.name: self.add_vault_data(snitch, snitch_secret_path)
+            for cluster in clusters
+            if (snitch := snitches_with_cluster_mapping.get(cluster.name))
+        }
+        return current_state
+
+    @staticmethod
+    def get_diff(
+        current_state: dict[str, Snitch], desired_state: list[ClusterV1]
```

Comment:
i find it confusing that `current_state` and `desired_state` are of different types.

Reply:
So actually the main reson they both are different is because of the `check_in_url`. having same type only suits in case of delete and update as we have a valid value for check_in_url` in that case. But in case of create we do not have a value for `check_in_url`. So I kept it different.